src/components/Graph.tsx

```diff
@@ -115,6 +120,23 @@ const Graph: React.FC<GraphProps> = ({ data, style = {}, focusedNodeID }) => {
     const instance = echarts.getInstanceByDom(chartDOM as any);
     if (instance) {
       instance.setOption(option);
+      instance.on('mouseover', function (param: any) {
+        if (param.dataType === 'node') {
+          const nodeDataIndex = param.data.id;
+          if (onToolTipChange) {
+            onToolTipChange(nodeDataIndex, true);
+          }
+        }
+      });
+      instance.on('mouseout', function (param: any) {
+        if (param.dataType === 'node') {
+          const nodeDataIndex = param.data.id;
+          if (onToolTipChange) {
+            onToolTipChange(nodeDataIndex, false);
+          }
```

Comment:
This part can be simplified into one single line:  `onToolTipChange && onToolTipChange(param.data.id, false);` Same for the mouseover.